openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/LightGBMResultTest.java

```diff
@@ -0,0 +1,276 @@
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static java.nio.file.Files.createTempFile;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+/**
+ * Test class to ensure that Microsoft LightGBM predicts the expected
+ * <i>class distribution</i> and <i>feature contributions</i>.
+ *
+ * @author Sheng Wang (sheng.wang@feedzai.com)
+ * @since 1.2.2
+ */
+public class LightGBMResultTest {
+
+    /**
+     * Logger for this class.
+     */
+    private static final Logger logger = LoggerFactory.getLogger(LightGBMResultTest.class);
+
+    /**
+     * Parameters for model training.
+     */
+    private static final Map<String, String> MODEL_TRAINING_PARAMS = getTestParams();
+
+    /**
+     * Result Schema:
+     * <ul>
+     *     <li>card -> number</li>
+     *     <li>amount -> number</li>
+     *     <li>cat1_generator -> number</li>
+     *     <li>cat2_generator -> number</li>
+     *     <li>cat3_generator -> number</li>
+     *     <li>num1_float -> number</li>
+     *     <li>num2_float -> number</li>
+     *     <li>num3_float -> number</li>
+     *     <li>fraud_label -> number</li>
+     * </ul>
+     */
+    public static final DatasetSchema SCHEMA = getTestSchema();
+
+    /**
+     * Result Schema:
+     * <ul>
+     *     <li>card -> number</li>
+     *     <li>amount -> number</li>
+     *     <li>cat1_generator -> number</li>
+     *     <li>cat2_generator -> number</li>
+     *     <li>cat3_generator -> number</li>
+     *     <li>num1_float -> number</li>
+     *     <li>num2_float -> number</li>
+     *     <li>num3_float -> number</li>
+     *     <li>bias -> number</li>
+     *     <li>predictions -> number</li>
+     * </ul>
+     */
+    public static final DatasetSchema RESULT_SCHEMA = getTestResultSchema();
+
+    /**
+     * The max number of instances to train, i.e., the number of lines in the file {@link treeshap_t treeshap_train.csv}.
+     */
+    public static final int MAX_INSTANCES_TO_TRAIN = 40000;
+
+    /**
+     * The chunk size instances, i.e., the number of chunk instances it keeps in memory.
+     */
+    public static final int CHUNK_SIZE_INSTANCES = 4;
+
+    /**
+     * Gets test schema.
+     *
+     * @return a {@link DatasetSchema}.
+     */
+    private static DatasetSchema getTestSchema() {
+        final List<FieldSchema> schema = new ArrayList<>(9);
+        addFields(schema);
+        schema.add(new FieldSchema("fraud_label", 8, new NumericValueSchema(false)));
+
+        return new DatasetSchema(
+                8,
+                schema
+        );
+    }
+
+    /**
+     * Gets test result schema.
+     *
+     * @return a {@link DatasetSchema}.
+     */
+    private static DatasetSchema getTestResultSchema() {
```

Comment:
What is this test result schema? A table with the statistics of the results? If so, please write it down